ArjunaJTA/jms/src/main/java/org/jboss/narayana/jta/jms/JmsXAResourceRecoveryHelper.java

```diff
@@ -177,4 +268,11 @@ private void disconnect() throws XAException {
         }
     }
 
+    private void assertIfRecoveryScanWasStarted() throws XAException {
+        if (delegate == null) {
+            LOGGER.warn("XAResource.recover has to be called first with XAResource.TMSTARTRSCAN argument");
+            throw new XAException(XAException.XAER_PROTO);
```

Comment:
With XAException it's possible to happen. If connect() method fails, XARecoveryModule#xaRecoveryFirstPass will catch the exception and will then call XAResource.TMENDRSCAN which will fail because of not available connection. NPE however, should be populated to XARecoveryModule#periodicWorkFirstPass once the connect() fails and only end up being logged and no XAResource.TMENDRSCAN should be executed.